src/CabanaPD_Solver.hpp

```diff
@@ -580,6 +568,51 @@ class SolverFracture
         this->final_output();
     }
 
+    void run()
+    {
+        this->init_output( 0.0 );
+
+        // Main timestep loop.
+        for ( int step = 1; step <= num_steps; step++ )
+        {
+            _step_timer.start();
+
+            // Integrate - velocity Verlet first half.
+            integrator->initialHalfStep( *particles );
+
+            if constexpr ( std::is_same<typename force_model_type::thermal_type,
+                                        TemperatureDependent>::value )
+                comm->gatherTemperature();
+
+            // Update ghost particles.
+            comm->gatherDisplacement();
+
+            // Update force and related quantities.
+            updateForce();
```

Comment:
Bond-breaking is inside the force computation, right? 

Note we need to ensure all bonds break before the force computation, to ensure symmetry of bond-breaking. 

For example, in PDMATLAB2D we have for time integrator:

 % Step 1: Compute velocity at t+dt/2 for all nodes 

% Step 2: Compute displacement at t+dt for all nodes  

**% Update neighbor list if bond breaking occurs**

**% Compute internal force density and macroelastic energy density at t+dt for all nodes**

 % Step 3: Compute velocity at t+dt for all nodes

Reply:
I think this is unrelated - maybe open this as a separate issue